### PR TITLE
Twilson63/feat connect nodejs nodejs 372

### PR DESCRIPTION
**Pull request by twilson63**
This PR includes the refactored versions of hyper-connect for deno and nodejs, the nodejs package handles four different NodeJS environments, TS, ESM, CJS, and VITE/esbuild 

Deno mirrors the node version but is located in the `deno` folder.

All Tests are passing and both build tools compile.

### CODE REVIEW

File: packages/connect/README.md [hunk 1]
```diff
@@ -19,91 +20,106 @@ INSERT BADGES HERE
...
+await hyper.data.add({ id: "game-3", type: "game", name: "Galaga" });
 
-const result = await hyper.search.query(criteria);
+const results = await hyper.data.query({ type: "game" });
 ```
 
----
+With hyper-connect you can access all of the hyper services and get back a
+Request object via a promise, then you can add that request object to your fetch
+call to invoke your request and receive a response.
```
Comment (TillaTheHun0) [inline packages/connect/README.md:93]: The wording here seems strange. Maybe something like:

> With `hyper-connect`, you can access all of the hyper services. Each `hyper-connect` api returns a `Request` object, via a `Promise`. You can then pass this `Request` object to your favorite http client ie. `fetch`.

File: packages/connect/README.md [hunk 2]
```diff
@@ -19,91 +20,106 @@ INSERT BADGES HERE
...
 };
 
-const req = await hyper.data.add(doc);
-const response = await fetch(req);
-if (response.ok) {
-  const result = await response.json();
-  console.log(result);
-}
+const result = await hyper.data.add(doc);
+console.log(result); // {ok: true, id: "movie-1"}
```
Comment (TillaTheHun0) [inline packages/connect/README.md:108]: These examples use the proxy, but the description above details using `hyper-connect` _along_ with fetch. Perhaps i'm misunderstanding.

File: packages/connect/deno/mod.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    return new Request(url, options);
+  };
+
+  const handleResponse = (response: Response) =>
+    Promise.resolve(response)
+      .then(
+        ifElse(
+          (r: Response) =>
+            r?.headers?.get("content-type")
+              ? r?.headers?.get("content-type")?.includes("application/json")
```
Comment (TillaTheHun0) [inline packages/connect/deno/mod.ts:29]: Are all of these `?` necessary? if `r` is of type `Response`, I think all of these shouldn't be needed. If they're needed, then the type for `r` is incorrect and should be fixed. Can probably be in a separate PR though.

**Review by TillaTheHun0 — APPROVED** [review]
Just some suggestions on docs, and a question on some typescript optional chaining.

It would be nice to have a single codebase, or a "core" like you mentioned, but this gets us what we need to now. We can iterate on it and make it better.

File: packages/connect/deno/mod.ts (hunk 3 above)
Comment (twilson63) [reply packages/connect/deno/mod.ts:29]: Yeah, just trying to get TS to not be mad, may use a Ramda path here

**Comment by twilson63** [comment]
> Just some suggestions on docs, and a question on some typescript optional chaining.
> 
> 
> 
> It would be nice to have a single codebase, or a "core" like you mentioned, but this gets us what we need to now, and to iterate on.

I hear you, but there are a lot on inconsistencies between TS Deno and TS Node

Maybe dropping back to JS and using JSDocs to expose an index.d.ts file externally could work

Idk, I think this is progress and will continue to iterate
